jumanji/environments/routing/macvrp/env.py

```diff
@@ -0,0 +1,614 @@
+            max_demand: maximum demand of each node. Defaults to 10.
+            reward_fn: `RewardFn` whose `__call__` method computes the reward of an environment
+                transition. The function must compute the reward based on the current state,
+                the chosen action, the next state and whether the action is valid.
+                Implemented options are [`DenseReward`, `SparseReward`]. Defaults to `DenseReward`.
+            viewer: `Viewer` used for rendering. Defaults to `CVRPViewer` with "human" render mode.
+        """
+
+    def __init__(
+        self,
+        num_customers: int = 6,
+        num_vehicles: int = 2,
+        render_mode: str = "human",
+        reward_fn: Optional[RewardFn] = None,
+        viewer: Optional[Viewer] = None,
+    ):
+        """
+        Instantiates a `MACVRP` environment.
+
+        Args:
+            num_customers: number of customer nodes in the environment. Defaults to 20.
+            num_vehicles: number of vehicles in the environment. Defaults to 2.
+            render_mode: render mode used by the viewer. Defaults to "human".
+            reward_fn: `RewardFn` whose `__call__` method computes the reward of an environment
+                transition. The function must compute the reward based on the current state
+                and whether the environment is done.
+                Implemented options are [`DenseReward`]. Defaults to `DenseReward`.
+            viewer: `Viewer` used for rendering. Defaults to `MACVRPViewer` with "human" render
+                mode.
+        """
+
+        self.num_customers = num_customers
+        self.num_vehicles = num_vehicles
+
+        # Scenario are taken from the paper
+
+        # From the paper: All distances are represented by Euclidean distances in the plane,
+        # and the speeds of all  vehicles are assumed to be identical (i.e., it takes one
+        # unit of time to travel one unit of distance)
+        self.speed = 1
+
+        # Note: The time window detail could not be found in the paper for
+        # the 20, 50 and 150 customer scenarios. We use the 150 customer scenario's
+        # time window of 20 for them.
+        self.time_window_length = 20
+
+        (
+            self.map_max,
+            self.max_capacity,
+            self.max_start_window,
+            self.early_coef_rand,
+            self.late_coef_rand,
+            self.customer_demand_max,
+        ) = get_init_settings(num_customers, num_vehicles)
+
+        self.max_end_window = self.max_start_window + self.time_window_length
+
+        self.max_local_time = (
+            2.0 * jax.numpy.sqrt(2.0) * self.map_max * self.num_customers
+        )
+
+        self.reward_fn = reward_fn or DenseReward(
+            self.num_vehicles, self.num_customers, self.map_max
+        )
+
+        # Create viewer used for rendering
+        self._env_viewer = viewer or MACVRPViewer(
```

Comment:
NIT can we call this `_viewer`?
```suggestion
        self._viewer = viewer or MACVRPViewer(
```

Reply:
Updated :+1: